VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -438,6 +439,9 @@ public void onSaveInstanceState(Bundle outState) {
      */
     public synchronized void play(String path, int seekTo, String title, 
             TranscriptModel trModel, DownloadEntry video) {
+        // clear the window flag
+        setScreenOnWhilePlaying(false);
+
```

Comment:
Yes this is needed. If we play a online video and while video is preparing to play, if we turn on the network, then player shows an error message and in this case screen should be allowed to be OFF.  So, we keep the screen ON only if video is playing.